subprojects/core/src/integTest/groovy/org/gradle/initialization/CalculateTaskGraphBuildOperationIntegrationTest.groovy

```diff
@@ -123,10 +145,165 @@ class CalculateTaskGraphBuildOperationIntegrationTest extends AbstractIntegratio
         taskGraphCalculations[0].result.requestedTaskPaths == [":build"]
         taskGraphCalculations[1].details.buildPath == ":"
         taskGraphCalculations[1].result.requestedTaskPaths == [":build"]
-        taskGraphCalculations[2].details.buildPath== ":b"
+        taskGraphCalculations[2].details.buildPath == ":b"
         taskGraphCalculations[2].result.requestedTaskPaths == [":compileJava", ":jar"]
     }
 
+    @ToBeFixedForInstantExecution
+    def "exposes plan details"() {
+        file("included-build").mkdir()
+        file("included-build/settings.gradle")
+        file("included-build/build.gradle") << """
+        apply plugin:'java-library'
+        group = 'org.acme'
+        version = '1.0'
+        """
+
+        file('src/main/java/org/acme/Library.java') << """
+        package org.acme;
+
+        class Library {
+        }
+        """
+        settingsFile << """
+            includeBuild 'included-build'
+        """
+
+        buildFile << """
+                apply plugin:'java-library'
+
+                dependencies {
+                    implementation 'org.acme:included-build:1.0'
+                }
+                task independentTask
+                task otherTask
+                task anotherTask
+                task firstTask
+                task secondTask
+                task lastTask
+                task someTask
+
+                someTask.dependsOn anotherTask
+                someTask.dependsOn otherTask
+                someTask.mustRunAfter firstTask
+                someTask.shouldRunAfter secondTask
+                someTask.finalizedBy lastTask
+        """
+        when:
+        succeeds('classes', 'independentTask', 'someTask')
+
+        then:
+        operations()[0].result.taskPlan.task.identityPath == [":compileJava", ":processResources", ":classes", ":independentTask", ":anotherTask", ":otherTask", ":someTask", ":lastTask"]
+        operations()[0].result.taskPlan.dependencies.identityPath as Set == [[":included-build:compileJava"], [], [":processResources", ":compileJava"], [":anotherTask", ":otherTask"]] as Set
```

Comment:
Why a set comparison here? Shouldn't we be tying the dependencies to the specific tasks? All the others are list comparisons.

Reply:
Also maybe `def taskPlan = ...` to tidy it up a bit?

Reply:
fixed